### PR TITLE
doc: convert links in links.txt to point to TechDocs

### PR DESCRIPTION
**Pull request by peknis**
Converted links as far as possible.
Some of them are only available in old releases
that are not included in TechDocs.

### CODE REVIEW

File: doc/nrf/links.txt [hunk 1]
```diff
@@ -258,7 +258,7 @@
 .. _`nRF Connect SDK v0.3.0 documentation`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/0.3.0/nrf/index.html
 .. _`nRF Connect SDK latest documentation`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/index.html
 
-.. _`known issues page on the main branch`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/known_issues.html
+.. _`known issues page on the main branch`: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/releases_and_maturity/known_issues.html
 .. _`known issues for nRF Connect SDK v2.6.0`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/known_issues.html?v=v2-6-0
```
[the remark this answers is not in the document]
Comment (peknis) [reply doc/nrf/links.txt:264]: The known issues links will be updated later as soon as we have a solution from Zoomin.

File: doc/nrf/links.txt [hunk 2]
```diff
@@ -294,7 +294,7 @@
 .. _`known issues for nRF Connect SDK v1.4.1`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/known_issues.html?v=v1-4-1
 .. _`known issues for nRF Connect SDK v1.4.0`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/known_issues.html?v=v1-4-0
 
-.. _`connecting to nRF Cloud`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/networking/nrf_cloud.html#connecting
+.. _`connecting to nRF Cloud`: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/libraries/networking/nrf_cloud.html#connecting
 
 .. _`API documentation`:
```
Comment (greg-fer) [inline doc/nrf/links.txt:299]: For future reference: links to ZBOSS external docs won't be updated because we're just hosting DSR's docs.

Comment (peknis) [reply doc/nrf/links.txt:299]: Looks like I had this correct already. Nothing to fix.

Comment (peknis) [reply doc/nrf/links.txt:299]: Looks like I had this correct already.

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Comment by b-gent** [comment]
The links look correct but this will unfortunately require a rebase. Take your version as a base and manually add my changes in links.txt made in this PR -> https://github.com/nrfconnect/sdk-nrf/pull/14960/files#diff-0f07293a1aa5ec4d690d216efeec811821b4e039a8cf3c4272f4197e29eebc9d

**Review by b-gent — APPROVED** [review]

**Pull request merged**